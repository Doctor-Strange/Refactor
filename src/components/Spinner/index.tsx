import React from "react";

const Spinner = (props: ISpinner) => {
  return (
    <span
      style={{
        display: props.display,
        borderRightColor: props.color,
        width: props.width + "px",
        height: props.width + "px",
      }}
      className="Loading"
    ></span>
  );
};
interface ISpinner {
  display: string;
  width: number;
  color: string;
}

export default Spinner;
