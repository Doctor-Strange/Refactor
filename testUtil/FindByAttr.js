

export const FindByAttr = (wrapper, value) => {
  return wrapper.find(`[data-test="${value}"]`);
};
