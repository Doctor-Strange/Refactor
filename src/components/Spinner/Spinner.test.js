import React from "react";
import { mount } from "enzyme";
import { FindByAttr } from "../../../testUtil/FindByAttr";
import Spinner from "./";

const setup = (props) => {
  return mount(<Spinner {...props} />);
};

const props = {
  display: "block",
  width: 20,
  color: "#0099ff",
};

describe("Spinner component", () => {
  describe("snapshot", () => {
    test("/components/Spinner", () => {
      const wrapper = setup(props);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe("props", () => {
    test("display props", () => {
      const wrapper = setup(props);
      const span = FindByAttr(wrapper, "span");
      expect(span.instance().style.display).toEqual("block");
    });
  });
});
