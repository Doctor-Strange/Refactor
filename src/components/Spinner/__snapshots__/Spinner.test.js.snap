// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component snapshot /components/Spinner 1`] = `ReactWrapper {}`;
